### PR TITLE
feat(heatmap_visualizer): add heatmap visualizer

### PR DESCRIPTION
**Pull request by ktro2828**
Signed-off-by: Kotaro Uetake kotaro.uetake@tier4.jp

## Description

Add heatmap visualizer to visualize heatmap of detected objects' position 

heatmap_xx1_nishishinjuku_ALL1

## Tests performed

After starting autoware, run this command
```
ros2 launch heatmap_visualizer heatmap_visualizer.launch.xml input/objects:=perception/object_recognition/detection/objects
```

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [ ] The PR follows the [pull request guidelines].
- [ ] The PR has been properly tested.
- [ ] The PR has been reviewed by the code owners.

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [ ] There are no open discussions or they are tracked via tickets.
- [ ] The PR is ready for merge.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

File: perception/heatmap_visualizer/README.md [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+
+### How to run
+
+```shell
+ros2 launch heatmap_visualizer heatmap_visualizer.launch.xml input/objects:=<DETECTED_OBJECTS_TOPIC>
+```
+
+## Inner-workings / Algorithms
+
+In this implementation, add position of detected objects to occupancy grid map and rescale values to integer [0, 100] after normalizing to float [0, 1]. After publishing, the rescaled values are de-normalized to original values.
```
Comment (yukkysaito) [inline perception/heatmap_visualizer/README.md:15]: Pls describe more details about algorithm :pray:

Comment (yukkysaito) [reply perception/heatmap_visualizer/README.md:15]: Please also describe the purpose and when this can be used.

**Comment by ktro2828** [comment]
@yukkysaito I'm sorry to late to update description of this package. Please re-review.

**Comment by yukkysaito** [comment]
@ktro2828 
I have some questions :pray: 
- What do heat maps mean? (existence probability?)
- Can it subscribe to any topic of DetectedObjects type like a clustering output?
- When should we use this?

**Comment by ktro2828** [comment]
@yukkysaito Thanks for quick reply!

> What do heat maps mean? (existence probability?)

We create heatmap by accumulating poistions of bounding boxes on each gird map cell, by defaults. However, we have option to create heatmap from existence probability, it can be selected with `use_confidence`.

> Can it subscribe to any topic of DetectedObjects type like a clustering output?

Currently, No. It only can subscribe `autoware_auto_msgs.msg.DetectedObjects`. Should we support the other msg types, like `tier4_perception_msgs.mgs.DetectedObjectsWithFeature`?

> When should we use this?

This package is assumed to use for detection trend analysis of the detector. For instance, the heatmap shows "this detector can perform good for the objects near arround of our vehicle, but for far arround is bad".

**Review by yukkysaito — APPROVED** [review]
LGTM

**Pull request merged**